src/libraries/System.Formats.Tar/tests/TarReader/TarReader.GetNextEntry.Tests.cs

```diff
@@ -250,5 +250,43 @@ public void GetNextEntry_UnseekableArchive_ReplaceDataStream_ExcludeFromDisposin
                 Assert.Equal("Substituted", streamReader.ReadLine());
             }
         }
+
+        [Theory]
+        [InlineData(512)]
+        [InlineData(512 + 1)]
+        [InlineData(512 + 512 - 1)]
+        public void BlockAlignmentPadding_DoesNotAffectNextEntries(int contentSize)
+        {
+            byte[] fileContents = new byte[contentSize];
+            Array.Fill<byte>(fileContents, 0x1);
```

Comment:
This is good. A 0x1 char is not a valid char in any of the header fields, so if the archive is not reading entries from the correct position, this char would cause an exception when collecting the header fields.